src/inference/gibbs.jl

```diff
@@ -83,11 +83,24 @@ metadata(t::GibbsTransition) = (lp = t.lp,)
 DynamicPPL.getlogp(t::GibbsTransition) = t.lp
 
 # extract varinfo object from state
-getvarinfo(state) = state.vi
-getvarinfo(state::AbstractVarInfo) = state
+"""
+    gibbs_varinfo(model, sampler, state)
+
+Return the variables corresponding to the current `state` of the Gibbs component `sampler`.
+"""
+gibbs_varinfo(model, sampler, state) = varinfo(state)
+varinfo(state) = state.vi
+varinfo(state::AbstractVarInfo) = state
 
-# update state with new varinfo object
-gibbs_update_state(state::AbstractVarInfo, varinfo::AbstractVarInfo) = varinfo
+"""
+    gibbs_state(model, sampler, state, varinfo)
+
+Return the state of the Gibbs component with updated variables `varinfo`.
+
+The Gibbs component samples from the `model` using the provided `sampler` and its current
+state is `state`.
```

Comment:
```suggestion
The Gibbs component samples from the `model` using the provided `sampler` with the returned state.
```
?

Reply:
I wanted to say that the current state of the sampler, as computed in the previous iteration, is `state`. However, the function updates it by taking into account the new variables sampled by the other samplers.

Reply:
Hmm, how about:
```julia
Computes the new state, taking into account the new variables sampled by other Gibbs components.

# Arguments
- `model`: model targeted by the samplers.
- `sampler`: the sampler for this Gibbs component.
- `state`: the current state of `sampler`, as computed in the previous iteration.
- `varinfo`: `VarInfo` for this sampler.

```